MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/snapshot/SnapshotShareActivity.java

```diff
@@ -0,0 +1,176 @@
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.geometry.LatLngBounds;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.snapshotter.MapSnapshot;
+import com.mapbox.mapboxsdk.snapshotter.MapSnapshotter;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class SnapshotShareActivity extends AppCompatActivity {
+  private MapView mapView;
+  private MapSnapshotter mapSnapshotter;
+  private MapboxMap mapboxMap;
+  private FloatingActionButton cameraFab;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_snapshot_share);
+
+    cameraFab = findViewById(R.id.camera_share_snapshot_image_fab);
+    cameraFab.setImageResource(R.drawable.ic_camera);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+
+    // Set a callback for when MapboxMap is ready to be used
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(final MapboxMap mapboxMap) {
+
+        SnapshotShareActivity.this.mapboxMap = mapboxMap;
+
+        // When user clicks the map, start the snapshotting process with the given parameters
+        cameraFab.setOnClickListener(new View.OnClickListener() {
+          @Override
+          public void onClick(View view) {
+
+            Toast.makeText(SnapshotShareActivity.this, R.string.loading_snapshot_image, Toast.LENGTH_LONG).show();
+            startSnapShot(
+              mapboxMap.getProjection().getVisibleRegion().latLngBounds,
+              mapView.getMeasuredHeight(),
+              mapView.getMeasuredWidth());
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Creates bitmap from given parameters, and creates a notification with that bitmap
+   *
+   * @param latLngBounds of map
+   * @param height       of map
+   * @param width        of map
+   */
+  private void startSnapShot(LatLngBounds latLngBounds, int height, int width) {
+    if (mapSnapshotter == null) {
+      // Initialize snapshotter with map dimensions and given bounds
+      MapSnapshotter.Options options =
+        new MapSnapshotter.Options(width, height).withRegion(latLngBounds).withStyle(mapboxMap.getStyleUrl());
+
+      mapSnapshotter = new MapSnapshotter(SnapshotShareActivity.this, options);
+    } else {
+      // Reuse pre-existing MapSnapshotter instance
```

Comment:
Else should handle the case when mapSnapshotter is nont null but it did not finish yet doing the snapshot. Otherwise we keep on hitting the Illegal Stae Exception